### ISSUE TITLE
Build failure: python312packages.bsddb3

### ISSUE THREAD

**Issue by D3vil0p3r** [NixOS/nixpkgs#308232]
### Steps To Reproduce

1. build bsddb3 Python312packages

### Build log

```
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
Traceback (most recent call last):
  File "/build/bsddb3-6.2.9/test.py", line 42, in <module>
    import test3 as test
  File "/build/bsddb3-6.2.9/test3.py", line 138, in <module>
    from distutils.util import get_platform
ModuleNotFoundError: No module named 'distutils'
error: builder for '/nix/store/lwi0kygzgr0azj1hahcl57hzry3i5cw2-python3.12-bsddb3-6.2.9.drv' failed with exit code 1;
       last 10 log lines:
       > patching script interpreter paths in /nix/store/fidk6hgvsza9gl562y3vcpv2zrvcxhfs-python3.12-bsddb3-6.2.9-dist
       > Executing pythonRemoveTestsDir
       > Finished executing pythonRemoveTestsDir
       > Running phase: installCheckPhase
       > Traceback (most recent call last):
       >   File "/build/bsddb3-6.2.9/test.py", line 42, in <module>
       >     import test3 as test
       >   File "/build/bsddb3-6.2.9/test3.py", line 138, in <module>
       >     from distutils.util import get_platform
       > ModuleNotFoundError: No module named 'distutils'
       For full logs, run 'nix log /nix/store/lwi0kygzgr0azj1hahcl57hzry3i5cw2-python3.12-bsddb3-6.2.9.drv'.
```

### Additional context

Python 3.12 seems to have removed distutils package but it should be inside setuptools module according https://github.com/NixOS/nixpkgs/issues/253751

Sadly this package is a dependency of several Python packages.

### Notify maintainers

@jonringer @tjni @mweinelt 

PS: the package has not maintainers.

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.28, NixOS, 24.05 (Uakari), 24.05pre615148.6143fc5eeb9c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `"home-manager, nixos, nixpkgs"`
 - nixpkgs: `/home/athena/.nix-defexpr/channels/nixpkgs`
```
---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by mweinelt** [comment]
If bsddb imports distutils, then it is not compatible with Python 3.12. This is something for upstream to solve, but is unlikely to happen:

> This library is deprecated. If you are running Python 3 >= 3.6, please upgrade to [berkeleydb](https://pypi.org/project/berkeleydb/) library. Check [details](https://www.jcea.es/programacion/pybsddb.htm).

https://pypi.org/project/bsddb3/

**Comment by D3vil0p3r** [comment]
So all those packages that depend on this package, like python312packages.scrapy must be labeled as broken?

**Comment by mweinelt** [comment]
Scrapy claims to support Python 3.12. Are you sure bsddb3 is still in their dependency chain?

**Comment by D3vil0p3r** [comment]
> Scrapy claims to support Python 3.12. Are you sure bsddb3 is still in their dependency chain?

Im not sure about that.

**Comment by D3vil0p3r** [comment]
I see this merged PR https://github.com/NixOS/nixpkgs/pull/309729 that probably should solve the issue on scrapy... but about bsddb3, it must be removed from python 3.12 package group...

**Comment by AkechiShiro** [comment]
@mweinelt distutils is used in the test of bsddb3 to get the platform data, we patched this behavior in the following PR #308157

**Comment by AkechiShiro** [comment]
Isn't It better to patch the test to avoid using distutils but still having it run ?
